crates/swc_ecma_transforms_compat/src/es2017/async_to_generator.rs

```diff
@@ -459,6 +480,49 @@ impl VisitMut for Actual {
     fn visit_mut_stmts(&mut self, _n: &mut Vec<Stmt>) {}
 }
 
+struct ThisReplacer<'a> {
```

Comment:
this probably needs some :eye: , since it's copy-paste from arrow and maybe need to be reused instead of duplication.

Reply:
You would need `arguments` replaced too

Reply:
Would you mind elaborate? not sure if I correctly understand what need to be updated.

Reply:
You still need `ArgumentsReplacer` from `arrow`

Reply:
Yes, I aware of those - I was curious if it's related with current `this` replacement or you're thinking it's better to mimic what arrow does regardless since both are going through nearly same path. as discussed in the issue itself and wrote in here (`maybe need to be reused instead of duplication`) probabaly, eventually these two need to be consoliated though.

Reply:
IMO we could have a combined visitor for `this` and `arguments`, and use it both here and in arrow

Reply:
sounds like a considerable approach, but I'd probabaly suggest to tackle it as separate PR & issue since it'll contain bit out of the scope work for the current PR.

for those, I'm curious how to handle differences between 2 visitor, for example argreplacer does for `visit_mut_member_expr`, thisreplacer for `visit_mut_arrow_expr`. Anyway, I think it's worth to file an issue and track.

Reply:
`MemberExpr` doesn't need any special handle?

Reply:
I don't know, reason raised q. For those two visitor, one explicitly skips visit and declared as noop and one are not.

Reply:
Also I should correct myself for what I say about `ThisReplacer`, it is also needed in `ClassMethod` and `MethodProperty`

Reply:
Either cases I think we can take this discussion out of this pr.

Reply:
> it is also needed in ClassMethod

This pr is specifically for the classmethod.

Reply:
https://github.com/swc-project/swc/issues/2786

Reply:
I think `arguments` replacer should handle member expressions, otherwise `foo.arguments` will be replaced.
It sounds like a bug.